### ISSUE TITLE
Problems with equals method

### ISSUE THREAD

**Issue by mpeters-tha** [KilianB/JImageHash#46]
Hello,

I have an issue, not really a bug but maybe an improvement.
Here is the use case:
- I create a hash from a picture, then I store it on the disk as a String.
- I retrieve this string and I create a new Hash object.
- I run the equals method on hash from step 1 and 2, it returns false.

After investigation, I found it is because the following part in equals method:
		if (getClass() != obj.getClass())
			return false;
My Hash from step 1 is of class: com.github.kilianB.hashAlgorithms.DifferenceHash$DHash while the second one is of class: com.github.kilianB.hash.Hash.

I tried to cast the first one in (Hash) but it doesn't seems to view the object as it.

One way to solve this migh be to do something like this in equals method intead of using the .getClass():
		if (!(obj instanceof Hash))
			return false;

Cheers.
Matt

**Comment by KilianB** [comment]
Thanks for the comment. Would you be willing to create a small pull request? Then I can take a look at it and see if the change would break other places.

**Comment by XorgPoulotron** [comment]
Haha, I'm not devlopper, never done that, but I can try 👍

**Comment by XorgPoulotron** [comment]
Hi again, Sorry I need some help here. I've installed github desktop on my PC, cloned the repo, created a branch, edited the Hash.java file and commited the change, but when I want to publish my branch, it says "It looks like you don't have write access to kilianb / JImageHash". Did I miss something ? Should I create a fork instead, as proposed by gitHub ?

**Comment by KilianB** [comment]
Yes exactly, this is how git works . You create a fork (your own working copy of the code), which you are free to modify in any way you like. 

You then can make changes 

`git add . `
`git commit -m "message"`
`git push`

And afterwards create a pull request (this can be done via the github ui). Due to this process no one is able to change the code of a repository without permission of the actual owner but since your version is also online everyone can just download your version and try it out as well as be able to see who made the changes afterwards.

**Comment by XorgPoulotron** [comment]
Thanks. Done, Sorry for my newbeeness.
